### PR TITLE
This will allow the to delete a path a user created, fix some bugs and increment coverage

### PR DESCRIPTION
**Pull request by Jeremmmyyyyy**
- Now in the mytrips view we can delete a path on the DB by long clicking on it (NO WARNING SCREEN CURRENTLY IMPLEMENTED)

- Fix bugs

- Add tests

### CODE REVIEW

File: app/src/androidTest/java/com/example/triptracker/map/AddSpotNoCameraTest.kt [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+      TakePicture(
+          outputDirectory = file,
+          executor = cameraExecutor,
+          onCaptureClosedSuccess = {},
+          onCaptureClosedError = {},
+          context = mockContext)
+    }
+
+    // Check if the camera is displayed
```
Comment (SpaceMercury) [inline app/src/androidTest/java/com/example/triptracker/map/AddSpotNoCameraTest.kt:57]: maybe rephrase to check that the camera is not displayed, I understand it's what you meant just for easier readability

File: app/src/androidTest/java/com/example/triptracker/map/AddSpotNoCameraTest.kt
Comment (SpaceMercury) [inline app/src/androidTest/java/com/example/triptracker/map/AddSpotNoCameraTest.kt:1]: tests look good !

File: app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt [hunk 2]
```diff
@@ -148,6 +155,14 @@ fun DisplayItinerary(
                     // overview with the selected itinerary highlighted and the first pinned places
                     onClick()
```
Comment (SpaceMercury) [inline app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt:156]: Doesn't this one have to be removed if you have the combined clickable onClick? (see line 159)

**Review by SpaceMercury — APPROVED** [review]
Tests are great! The addition of the delete feature, even though it's a work in progress is a good step forward. Addition of the canBeDeleted parameter is good thinking ahead, to ensure we can only delete our trips from the profile screen. Great !

File: app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt (hunk 2 above)
Comment (Jeremmmyyyyy) [reply app/src/main/java/com/example/triptracker/view/home/HomeUtils.kt:156]: Yes in fact thanks !
